### ISSUE TITLE
#bug; Compound V3 - wrong wallet balances in subgraphs

### ISSUE THREAD

**Issue by Natanela** [messari/subgraphs#2363]
**Describe the bug**
TL;DR - wrong wallet balanced on Compound V3 subgraphs

We’ve identified several wallets with a wrong balance stored on the subgraph compared to on-chain.

**Reproduction**
For example, Ethereum [subgraph showing](https://api.thegraph.com/subgraphs/name/messari/compound-v3-ethereum/graphql?query=%7B%0A++accounts%28where%3A+%7Bid%3A+%220xab62b80fca2c60254978e0a2ee269c492f319b2d%22%7D%29+%7B%0A++++id%0A++++positions+%7B%0A++++++id%0A++++++side%0A++++++asset+%7B%0A++++++++id%0A++++++++symbol%0A++++++++lastPriceUSD%0A++++++++lastPriceBlockNumber%0A++++++++decimals%0A++++++%7D%0A++++++timestampClosed%0A++++++balance%0A++++++principal%0A++++++market+%7B%0A++++++++id%0A++++++++borrowIndex%0A++++++++supplyIndex%0A++++++++_baseTrackingBorrowSpeed%0A++++++++_baseTrackingSupplySpeed%0A++++++%7D%0A++++%7D%0A++%7D%0A%7D&variables=%22%7B%5Cn++%5C%22timestamp%5C%22%3A+1692127090%2C%5Cn++%5C%22limit%5C%22%3A+1000%5Cn%7D%22) wallet `0xab62b80fca2c60254978e0a2ee269c492f319b2d` has borrowed `1,463,004 USDC` without any collateral position, however when calling the `borrowBalanceOf` of the [USDC comet contract](https://etherscan.io/address/0xc3d688b66703497daa19211eedff47f25384cdc3) the borrow balance is `0`:
![Untitled](https://github.com/messari/subgraphs/assets/37778340/c8683590-480e-460c-a5f2-bdb964bd2e0e)

Additional account addresses with wrong balances:

Chain | Address
-- | --
Ethereum | 0x649cc3ff614a70741cc1ce099c0fa14bdc51ac11
Ethereum | 0xab62b80fca2c60254978e0a2ee269c492f319b2d
Ethereum | 0xdf7bdb2d8e3c2c39b9367d887d497d450ad1c597
Ethereum | 0x05e70979fd3355133f91fc7c6597a8274169b3be
Ethereum | 0x2178e1d614fefb2b304db58f07116d89f948fda1
Ethereum | 0xe7ce11f4b1cfd8e607b0ca64823ad645f260badc
Polygon | 0x19bf9810d6bfc4d1b8e742edb1448486b38dc35f
Polygon | 0x6822c1c54ed42ceb294a46b855de75e82f8f98c5
Polygon | 0x5d38341569a12d50a44cdce4643ca474073a9a8b
Polygon | 0x6d903cbfdc7cf765c13fa05241392c9c7c2f00b7

**Comment by bye43** [comment]
@dmelotik

**Comment by dmelotik** [comment]
I believe what is happening is that this [transaction](https://etherscan.io/tx/0x4037ef85d9971280af0770c9fd68b33b3b03ca6d6b375f1338c0a2a572867f14) repays all of the debt, and withdraws the remaining USDC supplied.

So it is only recorded as a withdraw, however the subgraph should also record it as a repay, or at least update the borrow position. Going to have to come back around when I have capacity for this.
